src/diagram.rs

```diff
@@ -472,6 +472,11 @@ impl Diagram {
     }
 }
 
+// thiserror has a special impl for backtrace which only works on nightly.
+// But we can disable this special impl by aliasing it.
+// See https://github.com/dtolnay/thiserror/issues/387#issuecomment-2473781561.
+type DefinitelyNotBacktrace = backtrace::Backtrace;
```

Comment:
Just making sure I understand, without this type alias the code will fail to compile on nightly? I assume it should compile without any issue on stable, right?

Also while this name is honestly very funny, we should rename it to something like `BacktraceAlias`.

Reply:
It's the other way round, it won't compile without this alias, `thiserror` only supports backtrace on nightly.

Reply:
I'm a little bit concerned about only including the file and line location since users won't always know which version of the source code to compare against. What if we add a `&str` message inside of `unknown_diagram_error!(~)` similar to `.expect(~)` to help narrow down which line had an error?

Reply:
On the other hand if we're only using this error type for truly unexpected errors, then I don't mind leaving it as-is for now and we can see if any of these errors actually occurs after exercising the feature for a while.